File: tzapi4android/src/main/java/com/tianzunchina/android/api/utils/image/CircleImageDrawable.java
package com.tianzunchina.android.api.utils.image;

import android.graphics.Bitmap;
import android.graphics.BitmapShader;
import android.graphics.Canvas;
import android.graphics.ColorFilter;
import android.graphics.Paint;
import android.graphics.PixelFormat;
import android.graphics.Shader;
import android.graphics.drawable.Drawable;

/**
 * 圆形图
 * 用法: iv.setImageDrawable(new CircleImageDrawable(bitmap));
 * CraetTime 2016-3-7
 * @author SunLiang
 */
public class CircleImageDrawable extends Drawable {

    private Bitmap mBitmap;
    private Paint mPaint;
    private int mWidth;

    public CircleImageDrawable(Bitmap mBitmap){
        this.mBitmap = mBitmap;
        BitmapShader bitmapShader = new BitmapShader(mBitmap, Shader.TileMode.CLAMP,
                Shader.TileMode.CLAMP);
        mPaint = new Paint();
        mPaint.setAntiAlias(true);
        mPaint.setShader(bitmapShader);
        mWidth = Math.min(mBitmap.getWidth(), mBitmap.getHeight());
    }

    public int getIntrinsicWidth() {
        return mWidth;
    }

    public int getIntrinsicHeight() {
        return mWidth;
    }

    @Override
    public void draw(Canvas canvas) {
        canvas.drawCircle(mWidth / 2, mWidth / 2, mWidth / 2, mPaint);
    }

    @Override
    public void setAlpha(int alpha) {
        mPaint.setAlpha(alpha);
    }

    @Override
    public void setColorFilter(ColorFilter colorFilter) {
        mPaint.setColorFilter(colorFilter);
    }

    @Override
    public int getOpacity() {
        return PixelFormat.TRANSLUCENT;
    }
}

File: tzapi4android/src/main/java/com/tianzunchina/android/api/utils/image/CircleImageDrawable.kt
package com.tianzunchina.android.api.utils.image

import android.graphics.Bitmap
import android.graphics.BitmapShader
import android.graphics.Canvas
import android.graphics.ColorFilter
import android.graphics.Paint
import android.graphics.PixelFormat
import android.graphics.Shader
import android.graphics.drawable.Drawable

/**
 * 圆形图
 * 用法: iv.setImageDrawable(new CircleImageDrawable(bitmap));
 * CraetTime 2016-3-7
 * @author SunLiang
 */
class CircleImageDrawable(private val mBitmap: Bitmap) : Drawable() {
    private val mPaint: Paint
    private val mWidth: Int

    init {
        val bitmapShader = BitmapShader(mBitmap, Shader.TileMode.CLAMP,
                Shader.TileMode.CLAMP)
        mPaint = Paint()
        mPaint.isAntiAlias = true
        mPaint.shader = bitmapShader
        mWidth = Math.min(mBitmap.width, mBitmap.height)
    }

    override fun draw(canvas: Canvas) {
        canvas.drawCircle((mWidth / 2).toFloat(), (mWidth / 2).toFloat(), (mWidth / 2).toFloat(), mPaint)
    }

    override fun getIntrinsicWidth(): Int {
        return mWidth
    }

    override fun getIntrinsicHeight(): Int {
        return mWidth
    }

    override fun setAlpha(alpha: Int) {
        mPaint.alpha = alpha
    }

    override fun setColorFilter(cf: ColorFilter?) {
        mPaint.colorFilter = cf
    }

    override fun getOpacity(): Int {
        return PixelFormat.TRANSLUCENT
    }
}
